crates/common/fmt/src/ui.rs

```diff
@@ -324,22 +457,24 @@ to                   {}
 transactionIndex     {}
 v                    {}
 value                {}{}",
-            self.block_hash.pretty(),
-            self.block_number.pretty(),
-            self.from.pretty(),
-            self.gas.pretty(),
-            self.gas_price.pretty(),
-            self.hash.pretty(),
-            self.input.pretty(),
-            self.nonce,
-            self.signature.map(|s| s.r.to_be_bytes_vec()).pretty(),
-            self.signature.map(|s| s.s.to_be_bytes_vec()).pretty(),
-            self.to.pretty(),
-            self.transaction_index.pretty(),
-            self.signature.map(|s| s.v).pretty(),
-            self.value.pretty(),
-            self.other.pretty()
-        )
+                    self.block_hash.pretty(),
+                    self.block_number.pretty(),
+                    self.from.pretty(),
+                    self.gas.pretty(),
+                    self.gas_price.pretty(),
+                    self.hash.pretty(),
+                    self.input.pretty(),
+                    self.nonce,
+                    self.signature.map(|s| s.r.to_be_bytes_vec()).pretty(),
+                    self.signature.map(|s| s.s.to_be_bytes_vec()).pretty(),
+                    self.to.pretty(),
+                    self.transaction_index.pretty(),
+                    self.signature.map(|s| s.v).pretty(),
+                    self.value.pretty(),
+                    self.other.pretty()
+                ),
+            _ => todo!("cast needs to be updated to support this new tx type")
```

Comment:
this would panic for L2 txs like deposit, and we need to handle this variant as well

Reply:
Ok, I went with the fallback approach in https://github.com/foundry-rs/foundry/commit/e06f98cc043b4fe2c8a47ac9dafedf272176c866, let me know if there's another way.